### PR TITLE
fix: package structure & type definition paths

### LINKED ISSUE #62: [guess-parser] has incomplete type definitions
Hi, thanks for publishing a new version of `guess-parser` to npm. It works great, when used in JavaScript code. However it can't be used as part of a TypeScript codebase. The type definition file `dist/guess-parser/index.d.ts` currently looks like this: 

```typescript
export { parseRoutes } from './src/parser';
export { detect } from './src/detector';
export { parseRoutes as parseAngularRoutes } from './src/angular';
export * from './src/react';
export * from './src/preact';
```

It refers to the `./src` directory which is not part of the npm package. I hope it is just a matter of configuring the build process. Unfortunately I don't really know how your build process works.

### PR DESCRIPTION
**Pull request by mgechev**
Fix #62 #65

Fixes only around the build setup - webpack and Lerna.

@KyleAMathews the Gatsby plugin may not be able to find `guess-webpack/api` with the latest published package. This PR should fix the problem. The issue was caused by my misunderstanding of what Lerna includes as package content.

Also, maybe it's worth it to publish `guess-api` as a separate package instead of relying on `guess-webpack/api`. This will simplify the build and the package semantics looks clearer.

### CODE REVIEW

File: packages/guess-webpack/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+api/index.js
```
Comment (mgechev) [inline packages/guess-webpack/.gitignore]: Publishing `guess-api` as a separate package will simplify things a bit and remove the need for `.gitignore` on level individual files.

File: packages/guess-webpack/webpack.config.js [hunk 2]
```diff
@@ -1,17 +1,10 @@
 const CopyWebpackPlugin = require('copy-webpack-plugin');
-const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
 
 const common = {
   mode: 'production',
   externals: [/^(@|\w).*$/i],
   resolve: {
-    extensions: ['.ts', '.tsx', '.js'],
-    plugins: [
-      new TsconfigPathsPlugin({
```
Comment (mgechev) [inline packages/guess-webpack/webpack.config.js]: We weren't using this plugin.

**Comment by KyleAMathews** [comment]
Either way could work as long as it's well documented. Since guess probably won't be used outside of webpack I assume, splitting them probably won't help much?

**Comment by dijs** [comment]
Separate packages seems cleaner, but for now this is great and would unblock me.

**Comment by mgechev** [comment]
I'll try it our again today with Gatsby and the demo project and if everything works as expected, I'll publish a release.

**Comment by addyosmani** [comment]
👍

**Pull request merged**